02-express-mongo-crud/intro-express/27-28/README.md

```diff
@@ -0,0 +1,113 @@
+- Express allows us to easily set up routes that will trigger actions such as rendering pages or returning JSON.
+
+### Hello World in Express
+
+```javascript
+var express = require('express');
+var app = express();
+
+app.get('/', function (req, res) {
+  res.send('Hello World!');
+})
+
+var server = app.listen(3000);
+```
+
+### Request Response Cycle
+
+Remember that the interwebs is many clients querying many servers. We've done a lot with clients and APIs, and now we want to write the server side code that handles the request and then responds with some data.
+
+![request](http://i.imgur.com/YXgj8.png)
+
+
+### Express file tree
+
+```
+├── server.js  // your server code
+├── package.json    // lists dependencies; changed by npm install --save somePackage
+├── public  // i.e. client-side
+│   ├── images  // images to serve to client
+│   ├── scripts
+│       └── app.js   // client-side javascript file
+│   └── styles
+│       └── style.css
+├── vendor // optional 2nd public dir for jQuery & bootstrap if we choose not to use CDNs
+├── views  // html files that we'll serve
+│   └── index.html
+└── node_modules  // don't edit files in here!
+    ├── express // etc
+```
+
+> We should also add `node_modules` to a `.gitignore` file so it is not checked into git.  Future developers can just run `npm install` to get the packages listed in `package.json`
+
+**In Express**
+
+Let's look at a basic `get` method in an express app.
+
+```js
+// server.js
+  var taquerias = [
+    { name: "La Taqueria" },
+    { name: "El Farolito" },
+    { name: "Taqueria Cancun" }
+  ]
+```
+
+```js
+  app.get('/api/taquerias', function (req, res) {
+    res.json(taquerias);
+  });
+```
+
+Note that the `app` object has a method called `.get()` which takes two arguments: a url and a callback function. The callback takes two arguments. The `req`, and `res`. These stand for "Request" and "Response" from the request response cycle. We'll be console logging these objects in the exercises.
+
+### Game Plan
+
+Today's <a href="https://github.com/SF-WDI-LABS/express-intro" target="_blank">exercises</a> are set up a bit like a tutorial to walk you through:
+
+  * creating a new project with Node and Express
+  * creating routes for clients to make requests to your server
+  * storing data on the server
+  * responding to GET requests with simple strings (`res.send`), or JSON data (`res.json`)
+  * serving static files (images, css...)
+
+
+### Docs, Resources, Further Reading
+
+1. <a href="http://expressjs.com/starter/installing.html" target="_blank">Starting an Express Project</a>
+2. <a href="http://expressjs.com/starter/hello-world.html" target="_blank">Express Hello World</a>
+3. <a href="http://expressjs.com/starter/static-files.html" target="_blank">Express Static Files</a>
+4. <a href="http://expressjs.com/4x/api.html#res.render" target="_blank">Express res.render()</a>
```

Comment:
maybe we don't need this; however maybe consider doing res.sendFile to send static HTML file.